docs/proposals/20250425-aro-hcp.md

```diff
@@ -0,0 +1,684 @@
+ ChannelGroup string `json:"channelGroup"`
+
+ // VersionGate requires acknowledgment when upgrading ARO-HCP y-stream versions (e.g., from 4.20 to 4.21).
+ // Allowed values are Acknowledge, WaitForAcknowledge and AlwaysAcknowledge default is WaitForAcknowledge.
+ VersionGate string `json:"versionGate"`
+
+
+ // IdentityRef is a reference to an identity to be used when reconciling the aro control plane.
+ // If no identity is specified, the default identity for this controller will be used.
+ IdentityRef *infrav1.AzureClusterIdentity `json:"identityRef,omitempty"`
+
+
+ // AdditionalTags are user-defined tags to be added to the Azure resources associated with the control plane.
+ AdditionalTags infrav1.Tags `json:"additionalTags,omitempty"`
+}
+
+
+// PlatformProfile represents the Azure platform configuration.
+type PlatformProfile struct {
+ // Location should be valid Azure location ex; centralus
+ Location string `json:"location,omitempty"`
+
+
+ // Resource group name where the Aro-hcp will be attached to it.
+ ResourceGroup string `json:"resourceGroup,omitempty"`
+
+
+ // ResourceGroup Ref name that is used to create the ResourceGroup CR. The ResourceGroupRef must be in the same namespace as the AroControlPlane and cannot be set with ResourceGroup.
+ ResourceGroupRef string `json:"resourceGroup,omitempty"`
+
+
+ // Azure subnet id
+ Subnet string `json:"subnet,omitempty"`
+
+
+ // Subnet Ref name that is used to create the VirtualNetworksSubnet CR. The SubnetRef must be in the same namespace as the AroControlPlane and cannot be set with Subnet.
+ SubnetRef string `json:"subnetRef,omitempty"`
+
+
+  // OutboundType represents a routing strategy to provide egress to the Internet. Allowed value is loadBalancer
+  OutboundType string `json:"outboundType,omitempty"`
+
+  // Azure Network Security Group ID
+  NetworkSecurityGroupID string `json:"networkSecurityGroupId,omitempty"`
+
+  // ManagedIdentities Azure managed identities for ARO HCP.
+  ManagedIdentities ManagedIdentities `json:"managedIdentities,omitempty"`
+}
+
+type ManagedIdentities struct {
+  // CreateAROHCPManagedIdentities is used to create the required ARO-HCP managed identities if not provided.
+  // It will create UserAssignedIdentity CR for each required managed identity. Default is false. 
+  CreateAROHCPManagedIdentities bool `json:"createAROHCPManagedIdentities,omitempty"`
+  
+ // ControlPlaneOperators Ref to Microsoft.ManagedIdentity/userAssignedIdentities
+ ControlPlaneOperators  *ControlPlaneOperators `json:"controlPlaneOperators,omitempty"`
+
+ // DataPlaneOperators ref to Microsoft.ManagedIdentity/userAssignedIdentities
+ DataPlaneOperators     *DataPlaneOperators `json:"dataPlaneOperators,omitempty"`
+
+ // ServiceManagedIdentity ref to Microsoft.ManagedIdentity/userAssignedIdentities
+ ServiceManagedIdentity string            `json:"serviceManagedIdentity,omitempty"`
+}
+
+type ControlPlaneOperators struct {
```

Comment:
Should we also list the exact role definitions each one of these will get?

Reply:
not sure if I understand, The field desc mentioning the role definition "control-plane" and similar for every UAMI

Reply:
Every control plane and data plane are assigned a specific Azure role definition(s) - [CNTRLPLANE-171](https://issues.redhat.com/browse/CNTRLPLANE-171). 

For example, the ingress managed identity is only assigned the "Azure Red Hat OpenShift Cluster Ingress Operator" role over the resource groups containing the VNET and the DNS Zone.

Reply:
okay, that is fine we will use the same role naming and keys. Would you mention the ref for it from the Hypershift [repo](https://github.com/openshift/hypershift) . I don't think the Jira url above is accessible for everyone.

Reply:
The mapping of control plane to role definition to resource group does not reside from the HyperShift repo. I am not sure if it is available in some public fashion on the ARO side 🤔

Reply:
I checked the hypershift repo, I don't think we need to set the Azure role definition, In CAPZ we will set the role key name similar to the example [here](https://github.com/Azure/ARO-HCP/blob/main/cluster-service/cluster-creation.md?plain=1#L175) ex; control-plane, cluster-api-azure, ingress, ..etc then hypershift will set the role definition. Hypershift has the role ref [here](https://github.com/openshift/hypershift/blob/main/support/config/constants.go#L93)

Reply:
it is [here](https://github.com/openshift-online/ocm-cluster-service/blob/main/pkg/clusterprovisioner/acm/aro/hosted_cluster_provision_step.go#L1078) in the ocm-cluster-service

Reply:
FWIW, we do not set the role definitions in HyperShift. That is assumed to be done prior to creating the HostedCluster.

The HyperShift link you mentioned is related to the HyperShift CLI, which does set the role definitions but that is only for development and testing use.

Reply:
hmm, so we need to have this defined some where yes/no ? 
Also this is more like implementation details for the UserAssignedIdentity CR that will be created we will need to set the right spec info

Reply:
I think it would be nice to be defined somewhere. 

Maybe it goes with the UserAssignedIdentity CR; is that defined somewhere?